project/src/main/java/com/google/sps/servlets/SearchDataServlet.java

```diff
@@ -0,0 +1,67 @@
+package com.google.sps.servlets;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ListMultimap;
+import com.google.sps.utilities.CommonUtils;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles search queries and addition of event keywords to index. */
+@WebServlet("/search-data")
+public class SearchDataServlet extends HttpServlet {
+  private ListMultimap<String, String> keywordToEventIds = ArrayListMultimap.create();
+  private static final String KEYWORD = "keyword";
+  private static final String EVENT_ID = "event-id";
+  private static final String DESCRIPTION = "description";
+  
+  /**
+   * Return search results for the keyword specified in the request.
+   *
+   * @param request servlet request
+   * @param response servlet response
+   * @throws IOException if Input/Output error occurs
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String keyword = request.getParameter(KEYWORD);
+    if (keyword == null) {
```

Comment:
maybe should just redirect to the main event discovery page? because in this case, wouldn't the user not be searching for anything

Reply:
Here, I am returning a JSON of the search results so I do not think I should redirect to the discovery page.